test/syscalls/linux/pty.cc

```diff
@@ -1148,6 +1148,23 @@ TEST_F(PtyTest, SwitchNoncanonToCanonNoNewlineBig) {
   ExpectFinished(replica_);
 }
 
+// If the canonical input buffer is empty, and we switch to non-canonical
+// mode, the input buffer should not be readable.
+TEST_F(PtyTest, SwitchCanonToNoncanonEmptyInput) {
+  constexpr char kInput[] = "";
+  ASSERT_THAT(WriteFd(master_.get(), kInput, sizeof(kInput)),
+              SyscallSucceedsWithValue(sizeof(kInput)));
+
+  DisableCanonical();
+
+  // Nothing available yet.
+  char buf[2] = {};
+  ASSERT_THAT(PollAndReadFd(replica_.get(), buf, /* read size */ 2, /* timeout */ 2),
```

Comment:
Can you rebase on top of https://github.com/google/gvisor/commit/b26cf20d40162945369121a5feaf6162c97bbdfb and use `kTimeoutShort`?

Reply:
done!